src/plugins/irc-events/mode.js

```diff
@@ -29,8 +29,19 @@ module.exports = function(irc, network) {
 			if (char === "k") {
 				targetChan.key = add ? mode.param : "";
 				client.save();
+			} else {
+				// CHANMODES[3] == channel modes that have no parameters
+				if (irc.network.options.CHANMODES[3].indexOf(char) > -1) {
+					targetChan.modes.push(char);
```

Comment:
This doesn't support removing modes?

Reply:
Does the `if` statement above not handle that?